Source/cryptalgo/SecureSocketPort.h

```diff
@@ -105,10 +105,14 @@ namespace Crypto {
 
             // Signal a state change, Opened, Closed or Accepted
             void StateChange() override {
+                if (IsOpen() && _context == nullptr) { // Initialize may not yet have happened
```

Comment:
By design the ASSERT(_context != nullptr) and lets add ASSERT(_ssl != nullptr) could be added here, cause if either of these are a nullptr, the Initialize should have signaled that by sending out an error != CORE_ERROR_NONE and than the statechange will never be called!
So I consider this unusefull additional code and think that it is a design flaw if a socket could be "open" but it was *not* Initialized(), see: https://github.com/rdkcentral/Thunder/blob/fe2b0a56aa5ab8e6d1bb1d75321931b5c4867e94/Source/core/SocketPort.cpp#L516